privacyidea/lib/audit.py

```diff
@@ -105,10 +106,20 @@ def search(config, param=None, user=None):
     if "timelimit" in param:
         timelimit = parse_timedelta(param["timelimit"])
         del param["timelimit"]
+    if "hidden_columns" in param:
+        hidden_columns = param["hidden_columns"]
+        del param["hidden_columns"]
 
     pagination = audit.search(param, sortorder=sortorder, page=page,
                               page_size=page_size, timelimit=timelimit)
 
+    # delete hidden columns from response
+    if hidden_columns:
+        for audit_row in pagination.auditdata:
+            for col in hidden_columns:
+                if col in audit_row:
```

Comment:
This looks strange.
Would there be a more elegant way with using dict comprehension or do we create a new object?

~~~~python
audit_row = { key: value for key, value in audit_row.items() if key not in hidden_columns }
~~~~

Reply:
that's nice indeed.